### PR TITLE
Add cop to enfore refute_empty over refute(actual.empty?)

### PR DESCRIPTION
**Pull request by abhaynikam**
PR adds cops to enforce the use of refute_empty over refute(actual.empty?)

Ref: https://github.com/rubocop-hq/minitest-style-guide#refute-empty

### CODE REVIEW

File: test/rubocop/cop/minitest/refute_empty_test.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    assert_correction(<<~RUBY, @cop)
+      class FooTest < Minitest::Test
+        def test_do_something
+          refute_empty(somestuff, 'the message')
+        end
+      end
+    RUBY
+  end
+
+  def test_registers_offense_when_using_refute_with_empty_and_variable_message
```
Comment (bbatsov) [inline test/rubocop/cop/minitest/refute_empty_test.rb]: Aren't tests with variable/constant message redundant? The way I see it you only check for the argument count in the cop itself, so this is bound to work whatever you pass to it.

File: lib/rubocop/cop/minitest/refute_empty.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Check if your test uses `refute_empty` instead of `refute(actual.empty?)`.
```
Comment (bbatsov) [inline lib/rubocop/cop/minitest/refute_empty.rb]: I think you meant to write this the other way around.

**Comment by abhaynikam** [comment]
@bbatsov : Resolved the comments on this PR too. Please have a look.

File: lib/rubocop/cop/minitest/refute_empty.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # This cop enforces to use `refute_empty` instead of
+      # using `refute(actual.empty?)`.
```
Comment (bbatsov) [inline lib/rubocop/cop/minitest/refute_empty.rb]: actual -> object

File: test/rubocop/cop/minitest/refute_empty_test.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    assert_correction(<<~RUBY, @cop)
+      class FooTest < Minitest::Test
+        def test_do_something
+          refute_empty(somestuff)
+        end
+      end
+    RUBY
+  end
+
+  def test_registers_offense_when_using_refute_with_empty_and_string_message
```
Comment (bbatsov) [inline test/rubocop/cop/minitest/refute_empty_test.rb]: I guess "string message" can be shortened to just "message".

**Comment by abhaynikam** [comment]
@bbatsov : Rebased and resolved the comments. Please have a look. Thanks

**Comment by bbatsov** [comment]
Looks good. Thanks!

**Pull request merged**